pkg/kubelet/kubelet.go

```diff
@@ -1789,6 +1788,17 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && !kubetypes.IsStaticPod(pod) {
```

Comment:
There is a function `isInPlacePodVerticalScalingAllowed` that basically checks the same thing. Can we just reuse it?